src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -3372,7 +3370,7 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             }
 
             // produce the secondary output (ref assembly) if needed
-            if (emitSecondaryAssembly)
+            if (getMetadataPeStreamOpt is not null)
```

Comment:
This allowed me to avoid a nullable suppression on `getMetadataPeStreamOpt` below.